private/oci/sign_and_push.bzl

```diff
@@ -0,0 +1,98 @@
+load("@contrib_rules_oci//oci:defs.bzl", "oci_push")
+load("@contrib_rules_oci//cosign:defs.bzl", "cosign_attach", "cosign_sign")
+load("//private/pkg:oci_image_spdx.bzl", "oci_image_spdx")
+
+PUSH_AND_SIGN_CMD = """\
+repository="$(stamp "{REPOSITORY}")"
+tag="$(stamp "{TAG}")"
+[[ -n $EXPORT ]] && echo "$repository:$tag" >> $EXPORT
+"$(realpath {ATTACH_CMD})" --repository "$repository"
+"$(realpath {SIGN_CMD})" --repository "$repository" --key "$KEY" --attachment sbom
```

Comment:
Can we add keyless signing here too? I vaguely recall saying I didn't think people used it, but I think they actually do.

Reply:
by keyless, you mean using a KMS? if yes, that's the case here. it uses gcp's km same as the cloudbuild_cosign script did.

Reply:
sorry no I mean whats in: https://github.com/GoogleContainerTools/distroless/blob/main/.cloudbuild/cloudbuild.yaml#L46-L59

Reply:
cosign's signing using identity rather than keys.

Reply:
sure. i'll do that.